sdk-core-protos/build.rs

```diff
@@ -131,5 +136,58 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
             ],
         )?;
 
+    let file_path = out.join("temporal.api.common.v1.rs");
+    post_process_payload_file(file_path)?;
+
+    Ok(())
+}
+
+fn post_process_payload_file(file_path: PathBuf) -> Result<(), Box<dyn std::error::Error>> {
+    let file_contents: BufReader<File> = BufReader::new(File::open(file_path.clone())?);
+    let mut new_file_contents = String::new();
+
+    let mut found_payload = false;
+    let mut written = false;
+    for line in file_contents.lines() {
+        let line = line?;
+        if line.contains("pub struct Payload {") {
+            new_file_contents.push_str("#[prost(skip_debug)]\n");
+            found_payload = true;
+        }
+
+        // Write the current line to the new content
+        new_file_contents.push_str(&line);
+        new_file_contents.push('\n');
+
+        if found_payload && line == "}" && !written {
+            let debug_impl = r#"
+impl std::fmt::Debug for Payload {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
```

Comment:
Even if we do need to keep manual replacement, there's no need to inject this here. The debug impl can live in the same place the current Display impl is.

Reply:
Without Debug being in the generated files, we get an error with `derive(::prost::Message)`

```
`temporal::api::common::v1::Payload` doesn't implement `derive_more::Debug`
the trait `derive_more::Debug` is not implemented for `temporal::api::common::v1::Payload`, which is required by `&std::vec::Vec<temporal::api::common::v1::Payload>: derive_more::Debug`
add `#[derive(Debug)]` to `temporal::api::common::v1::Payload` or manually `impl derive_more::Debug for temporal::api::common::v1::Payload`
the trait `derive_more::Debug` is implemented for `std::vec::Vec<T, A>`
required for `std::vec::Vec<temporal::api::common::v1::Payload>` to implement `derive_more::Debug`
1 redundant requirement hidden
required for `&std::vec::Vec<temporal::api::common::v1::Payload>` to implement `derive_more::Debug`
required for the cast from `&&std::vec::Vec<temporal::api::common::v1::Payload>` to `&dyn derive_more::Debug`
```

Reply:
Ugh, that's annoying. I wonder exactly where that's coming from. It still does this even after you have the `Debug` impl in  `sdk-core-protos/lib.rs`? I'm really surprised by that because the generated code is literally inlined into that file, so it should be identical to having it written into the generated code

Reply:
I didn't try putting the `Debug` impl in `sdk-core-protos/lib.rs`, you're right, that resolves the error